crates/rome_cli/src/commands/ci.rs

```diff
@@ -2,14 +2,61 @@ use crate::{
     configuration::load_configuration, execute_mode, CliSession, Execution, Termination,
     TraversalMode,
 };
+use rome_service::configuration::{FormatterConfiguration, LinterConfiguration};
 use rome_service::workspace::UpdateSettingsParams;
 
 use super::format::apply_format_settings_from_cli;
 
 /// Handler for the "ci" command of the Rome CLI
 pub(crate) fn ci(mut session: CliSession) -> Result<(), Termination> {
     let mut configuration = load_configuration(&mut session)?;
-    apply_format_settings_from_cli(&mut session, &mut configuration)?;
+
+    let formatter_enabled = session
+        .args
+        .opt_value_from_str("--formatter-enabled")
+        .map_err(|source| Termination::ParseError {
+            argument: "--formatter-enabled",
+            source,
+        })?;
+
+    let linter_enabled = session
+        .args
+        .opt_value_from_str("--linter-enabled")
+        .map_err(|source| Termination::ParseError {
+            argument: "--linter-enabled",
+            source,
+        })?;
+
+    if let Some(formatter) = &mut configuration.formatter {
+        if let Some(formatter_enabled) = formatter_enabled {
+            formatter.enabled = formatter_enabled
+        }
+    } else {
+        configuration.formatter = Some(FormatterConfiguration {
+            enabled: formatter_enabled.unwrap_or(true),
+            ..FormatterConfiguration::default()
+        })
+    }
```

Comment:
Nit: Flipping the nesting may simplify some of the logic
```suggestion
	if let Some(formatter_enabled) = formatter_enabled {
	  if let Some(formatter) = &mut configuration.formatter {
	  	formatter.enabled = formatter_enabled;
	  } else {
		  configuration.formatter = Some(FormatterConfiguration {
    	        enabled: formatter_enabled,
        	    ..FormatterConfiguration::default()
	        })
        }
	}
```